samplePDF/samplePDFFDBase.cpp

```diff
@@ -788,6 +825,51 @@ void samplePDFFDBase::CalcXsecNormsBins(int iSample){
   return;
 }
 
+//LW 
+//Setup chosen oscillation calculator for each subsample
+//Default Baseline Implementation
+//Add your own implementation in experiment specific SamplePDF Class if necessary!!
+void samplePDFFDBase::SetupOscCalc(double PathLength, double Density)
+{
+
+  Beta=1;
+  useBeta=false;
+  applyBetaNue=false;
+  applyBetaDiag=false;
+
+  for (int iSample=0; iSample < (int)MCSamples.size(); iSample++) {
+#if defined (USE_PROB3) && defined (CPU_ONLY) 
+    MCSamples[iSample].Oscillator = new BargerPropagator();
+    MCSamples[iSample].Oscillator->UseMassEigenstates(false);
+    MCSamples[iSample].Oscillator->SetOneMassScaleMode(false);
+    MCSamples[iSample].Oscillator->SetWarningSuppression(true);
+#endif
+
+#if not defined (USE_PROB3)
```

Comment:
I don't know how I feel about `not defined USE_PROB3` being how USE_CUDAPROB should be used :smile: I think it makes it a little harder to follow to be honest but it's probably fine

Reply:
Discussed and will change in the future especially when we have >2 osc calculators